crates/goose/src/agents/sub_recipe_execution_tool/executor.rs

```diff
@@ -9,9 +9,38 @@ use crate::agents::sub_recipe_execution_tool::lib::{
 use crate::agents::sub_recipe_execution_tool::tasks::process_task;
 use crate::agents::sub_recipe_execution_tool::workers::{run_scaler, spawn_worker, SharedState};
 
-pub async fn execute_single_task(task: &Task, config: Config) -> ExecutionResponse {
+pub async fn execute_single_task(task: &Task, global_config: Config) -> ExecutionResponse {
     let start_time = Instant::now();
-    let result = process_task(task, config.timeout_seconds).await;
+
+    // Extract config from task payload if present and merge with global config
+    let effective_config = if let Some(task_config_value) = task.payload.get("config") {
```

Comment:
can we move this into a merge method on Config? that way if we add more fields, we are more likely to implement merging too. method should take another Config then I think